ironfish/src/rpc/routes/wallet/createTransaction.ts

```diff
@@ -0,0 +1,233 @@
+          })
+          .defined(),
+      )
+      .optional(),
+    burns: yup
+      .array(
+        yup
+          .object({
+            assetId: yup.string().defined(),
+            value: yup.string().defined(),
+          })
+          .defined(),
+      )
+      .optional(),
+    fee: yup.string().optional(),
+    feePriorityLevel: yup.string().optional(),
+    expiration: yup.number().nullable().optional(),
+    expirationDelta: yup.number().nullable().optional(),
+  })
+  .defined()
+
+export const CreateTransactionResponseSchema: yup.ObjectSchema<CreateTransactionResponse> = yup
+  .object({
+    transaction: yup.string().defined(),
+  })
+  .defined()
+
+router.register<typeof CreateTransactionRequestSchema, CreateTransactionResponse>(
+  `${ApiNamespace.wallet}/createTransaction`,
+  CreateTransactionRequestSchema,
+  async (request, node): Promise<void> => {
+    const transaction = request.data
+
+    const account = node.wallet.getAccountByName(transaction.fromAccountName)
+
+    if (!account) {
+      throw new ValidationError(`No account found with name ${transaction.fromAccountName}`)
+    }
+
+    // The node must be connected to the network first
+    if (!node.peerNetwork.isReady) {
+      throw new ValidationError(
+        `Your node must be connected to the Iron Fish network to send a transaction`,
+      )
+    }
+
+    if (!node.chain.synced) {
+      throw new ValidationError(
+        `Your node must be synced with the Iron Fish network to send a transaction. Please try again later`,
+      )
+    }
+
+    const receives = transaction.receives.map((receive) => {
+      let assetId = Asset.nativeId()
+      if (receive.assetId) {
+        assetId = Buffer.from(receive.assetId, 'hex')
+      }
+
+      return {
+        publicAddress: receive.publicAddress,
+        amount: CurrencyUtils.decode(receive.amount),
+        memo: receive.memo,
+        assetId,
+      }
+    })
+
+    const mints: MintDescription[] = []
+    if (transaction.mints) {
+      transaction.mints.map((mint) => {
+        mints.push({
+          asset: new Asset(mint.privateKey, mint.name, mint.metadata),
+          value: CurrencyUtils.decode(mint.value),
+        })
+      })
+    }
+
+    const burns: BurnDescription[] = []
+    if (transaction.burns) {
+      transaction.burns.map((burn) => {
+        let assetId = Asset.nativeId()
+        if (burn.assetId) {
+          assetId = Buffer.from(burn.assetId, 'hex')
+        }
+        burns.push({
+          assetId: assetId,
+          value: CurrencyUtils.decode(burn.value),
+        })
+      })
+    }
+
+    const totalByAssetIdentifier = new BufferMap<bigint>()
+    if (transaction.fee) {
+      const fee = CurrencyUtils.decode(transaction.fee)
+      if (fee < 1n) {
+        throw new ValidationError(`Invalid transaction fee, ${transaction.fee}`)
+      }
+
+      totalByAssetIdentifier.set(Asset.nativeId(), fee)
+    }
+
+    for (const { assetId, amount } of receives) {
```

Comment:
We iterate over the receives both here and on line 118. We could consolidate the two loops